### ISSUE TITLE
Functions to copy from Graphs to StatsGraph

### ISSUE THREAD

**Issue by harshitmotwani2015** [Macaulay2/Workshop-2020-Warwick#38]
1) Graphs Datatype
2) Digraphs datatype
3) Mixed graphs datatype
4) delete edges
5) delete vertices
6) descendants
7) sort vertices
8) parents 
9) non neighbours
10) bfs
11) dfs
12) topological sort
13) collate vertices
14) add edge
15) add vertices
16) top sort
17) isConnected
18) isCyclic
19) children

**Comment by olgakuznetsova** [comment]
@harshitmotwani2015 IMHO, some of these do not need to be copied but instead we need to include "PackageExports=>{"Graphs"}" in the preamble because we want StatGraphs to contain only the functions specific to statistical graphs. I am referring to:
- Graphs Datatype
- Digraphs datatype
- delete edges
- delete vertices
- descendants
- parents
- non neighbours
- bfs
- dfs
- add edge
- add vertices
- isConnected
- isCyclic
- children

Could you tell where to find "sort vertices". I can't seem to find it in the code for Graphs.m2 or GraphicalModels.m2

**Comment by olgakuznetsova** [comment]
Or did you mean to specialise those functions to MixedGraphs? (Like: https://github.com/Macaulay2/Workshop-2020-Warwick/blob/f1b48b2dc52bbfa33e1aab6f158e312983933ea6/AlgebraicStatistics/MLE/StatGraphs.m2#L84-L90)

**Comment by harshitmotwani2015** [comment]
I was thinking that we don't export graphs package at all and make our stats graph package totally independent from it by having those functions which we will need for graphical models and graphical models mle. And this list contains all the functions used from graphs package in graphical models. But we can also go according to your approach.

And it is sort vertices here https://github.com/Macaulay2/Workshop-2020-Warwick/blob/f1b48b2dc52bbfa33e1aab6f158e312983933ea6/AlgebraicStatistics/MLE/GraphicalModels.m2#L533

Its just a sorting function in M2. Sorry, I confused it with graphs function.

**Comment by olgakuznetsova** [comment]
@harshitmotwani2015  Thanks for the clarification! Yep, I think it's better to only keep the stuff that is unique to statistical graphs in the StatGraphs package for two reasons:

1. User might need to use both Graphs and StatGraphs packages in their code. Functions with the same name will create conflicts and break their code.
2. Graphs team might want to improve the code inside their functions. This will not be available in our package.

So, I think we want the packages to be complementary but not overlapping.

**Comment by harshitmotwani2015** [comment]
@olgakuznetsova Thank you very much for clarifications. I will then just keep those functions in stats graph which are highly specific to graphical models. But, I will keep this list as these are all the functions we need for graphical models.

**Comment by olgakuznetsova** [comment]
Thanks! And we may need to specialise some of the functions (e.g. we need to decide whether a user should be able to use add vertices directly with mixed graphs or perform that outside of this package)

**Comment by luisgarciapuente** [comment]
I completely agree with Olga’s suggestions.

Thanks

Luis

**Comment by luisgarciapuente** [comment]
I think that the actual list is:

  "topSort",
        "newDigraph",
        "SortedDigraph",
    "Bigraph",
    "bigraph",
    "LabeledGraph",
    "labeledGraph",
    "MixedGraph",
    "mixedGraph",
    "collateVertices"

**Comment by harshitmotwani2015** [comment]
Thank you very much @luisgarciapuente. I will add these functions to the stats graph package along with the necessary specializations as suggested by @olgakuznetsova.

**Comment by luisgarciapuente** [comment]
Update on Graphs.m2: I contacted the current list of maintainers and nobody replied. I tried several different email addresses since it seems that some of them have moved to other institutions. I also tried to contact some of the original authors. I only got one reply (from Amelia Taylor) essentially stating that we were free to make changes and add our names to the list of contributors of the package.
